File: OpenMPy/examples/Monte.py
from omp import *
import sys
import random
import math

numIters = 0

if len(sys.argv) != 2:
    print("usage: ", argv[0], " <num_iterations>")

try:
    numIters = int(sys.argv[1])
except ValueError:
    print("Cannot convert argument to integer")
    sys.exit(1)

numIn = 0
numOut = 0

#omp parallel num_threads(4) reduction(+:numIn) reduction(+:numOut)
    localRandom = random.Random()

    myIters = numIters
    numIn = 0
    numOut = 0
    #omp for
        for i in range(myIters):
            #get random number from 0 to 1
            x = localRandom.uniform(0,1)
            y = localRandom.uniform(0,1)
            hyp = math.sqrt(x*x + y*y)
            if hyp < 1.0:
                numIn += 1
            else:
                numOut += 1

p = float(numIn)/(numIn+numOut)
fourp = 4*p
print("Pi is ", fourp)

File: PyjamaTraining/src/Monte.java
import java.lang.*;
import java.lang.Math;
import java.util.Random;
import java.util.concurrent.ThreadLocalRandom;


public class Monte {
	public static void main(String[] args) {
		int numIters = 0;
		try {
			numIters = Integer.parseInt(args[0]);
		} catch (Exception ex) {
			System.err.println("Exception when parsing argument");
			System.exit(1);
		}

		int numIn = 0;
		int numOut = 0;
		//#omp parallel num_threads(4) shared(numIters) reduction(+:numIn) reduction(+:numOut)
		{
			int myIters = numIters;
			ThreadLocalRandom rand = ThreadLocalRandom.current();
			numIn = 0;
			numOut = 0;
			//#omp for
			for (int i = 0; i < myIters; i++) {
				// get random number from 0 to 1
				double x = rand.nextDouble();
				double y = rand.nextDouble();
				double hyp = Math.sqrt(x*x + y*y);
				if (hyp < 1.0) {
					numIn++;
				} else {
					numOut++;
				}
			}
		}
		float p = (((float)numIn)/(numIn+numOut));
		float fourp = 4*p;
		System.out.println("Pi is " + String.valueOf(fourp));
	}
}
